### PR TITLE
Changed internal bot to only listen on public rooms

### LINKED ISSUE #895: Bot should not track conversations on DM and Private Groups
This is a  screenshot from a Direct Message:

screen shot 2015-09-25 at 13 15 30

This is a screenshot from a Private Group
screen shot 2015-09-25 at 13 31 23

### PR DESCRIPTION
**Pull request by geekgonecrazy**
this will close #895

Currently the bot listens in every room.  When having a private conversation its really annoying to have the bot pop in.  Could probably be limited further.  But this to me is a great start.

### CODE REVIEW

**Comment by gmsecrieru** [comment]
:+1:

**Comment by geekgonecrazy** [comment]
@RocketChat/owners what say ye?  Is this an acceptable change?

**Comment by sampaiodiego** [comment]
I still would to be able to use bot on private rooms.
So, I think bot should answer only when mentioned on private rooms.

**Comment by geekgonecrazy** [comment]
@sampaiodiego hmm.. So we'd have to some how make most of the other scripts only respond if in public room.  I'm not sure how to implement that.

**Comment by sampaiodiego** [comment]
hmm.. I don't know if I typed what I wanted to. =P let me try again.

I think the following behavior is ok to me:

* when you're in a public channel, if you type `gob it`, there are a hubot script that will answer with a  GOB's image.
* when you're in 1-1 room, if type `gob it`, nothing happens.
* when you're in 1-1 room, if you type: `@rocketbot gob it`, rocketbot will answer with a GOB's image.

**Comment by geekgonecrazy** [comment]
oooh.  So this is gonna have to pretty much be done in each script.  o.O

**Comment by sampaiodiego** [comment]
I don't think so. you should have a `mentions` array on message's object.. we only have the change the if statment to something like this:

`if room.t is 'c' or message.mentions.indexOf('rocketbot') isnt -1`

Or am I wrong?

**Comment by graywolf336** [comment]
Would this also disable the direct messages to RocketBot? Because sometimes I use him via direct messaging him

**Comment by geekgonecrazy** [comment]
@sampaiodiego looks like mentions has object in it.  heh   But in theory that'd work.

@graywolf336 yeah... would need an exception for if direct and to rocketbot.

**Comment by rodrigok** [comment]
IMHO bot should never respond until mentioned: **bot doSomething**

So we need to disable scripts that don't require to call the bot

**Comment by geekgonecrazy** [comment]
@rodrigok That would be preferable.

That would completely remove the need for this PR.

But it is "neat"

**Comment by geekgonecrazy** [comment]
@rocketchat/owners what do we want to do with this?  I know we're doing a bot name change.  #921 do we want to cut any packages etc?

Should I just change this to public channels or direct msg to bot is all it listens on?

**Comment by Sing-Li** [comment]
Wait. We have roles and permissions now.  Should we just add option to ban bot in room?  And add bot role to all bot users.

**Comment by marceloschmidt** [comment]
Following @Sing-Li idea, we could probably have three permissions:

'voice-public': allows a user to speak in a channel
'voice-group': likewise for groups
'voice-private': likewise for private messages

Users without voice-[type] would not be able to speak on those type of rooms. So giving only voice-public to the 'bot' role would enable bots to speak only on public rooms... Was that your idea, @Sing-Li ?

**Comment by marceloschmidt** [comment]
Let's open a new issue to do that, if required, and merge this PR for now. I think we can start with what we have already.
